### PR TITLE
Regaddressfix

### PR DESCRIPTION
**Pull request by AstryX**
Whitelist fees set to zero. Fixed register address. for Multisig. Added a few tests and applied changes Nikos requested in the previous pull request. Refactored some code.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -577,6 +576,17 @@ static void SendOnboardTx(const CScript& script,  CWalletTx& wtxNew){
         CAsset assetid = out.tx->GetOutputAsset(out.i);
         if (nValue >= nAmount && assetid == whitelistAsset){
             coinControl->Select(COutPoint(out.tx->GetHash(), out.i));
+            assert(out.i < out.tx->tx->vout.size());
```
Comment (nkostoulas) [inline src/wallet/rpcwallet.cpp]: why is this `assert` needed

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -2855,6 +2855,8 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
             AvailableCoins(vAvailableCoins, true, coinControl);
 
 	    nFeeRet = 1;
+        if (feeAsset == whitelistAsset)
```
Comment (nkostoulas) [inline src/wallet/wallet.cpp]: Might be good to generalise this to all policy assets

Comment (AstryX) [reply src/wallet/wallet.cpp]: Has been applied to all policy assets

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (AstryX) [reply src/wallet/rpcwallet.cpp]: assert has been removed.

**Comment by AstryX** [comment]
Tested and ready to go

**Comment by AstryX** [comment]
> What's the purpose of changing set to vector? Is it the case we can the same pub key multiple times? Should that be expected?

Set was changed to a vector, as explained, to retain the order of the pubkeys which is essential in multisig  address recreation/verification. That data is not used for that purpose at the moment,  but there may be a time in the future when we will need it.

**Pull request merged**